File: python/src/coffeeshop_kata/status.py
from enum import Enum


class Status(Enum):
    """Enumeration for all the statuses an Order can be in."""
    
    PAYMENT_EXPECTED = "payment_expected"
    """Placed, but not paid yet. Still changeable."""
    
    PAID = "paid"
    """Order was paid. No changes allowed to it anymore."""
    
    PREPARING = "preparing"
    """The Order is currently processed."""
    
    READY = "ready"
    """The Order is ready to be picked up by the customer."""
    
    TAKEN = "taken"
    """The Order was completed."""

File: swift/Sources/CoffeeShopKata/Status.swift
import Foundation

/**
 * Enumeration for all the statuses an Order can be in.
 */
public enum Status: String, CaseIterable {
    /**
     * Placed, but not paid yet. Still changeable.
     */
    case paymentExpected = "PAYMENT_EXPECTED"
    
    /**
     * Order was paid. No changes allowed to it anymore.
     */
    case paid = "PAID"
    
    /**
     * The Order is currently processed.
     */
    case preparing = "PREPARING"
    
    /**
     * The Order is ready to be picked up by the customer.
     */
    case ready = "READY"
    
    /**
     * The Order was completed.
     */
    case taken = "TAKEN"
} 